Sources/View/EditView.swift

```diff
@@ -77,9 +79,30 @@ struct EditView: View {
         .padding(20)
     }
 
+    var search: View {
+        FormSection(Loc.searchTitle) {
+            Form {
+                EntryRow(Loc.searchFlashcards, text: Binding(
+                    get: {
+                        self.searchQuery
+                    },
+                    set: { newQuery in
+                        self.searchQuery = newQuery
+                    }
+                ))
+            }
+        }
+        .padding(20)
+    }
+
     var flashcards: View {
         ForEach(.init(set.flashcards.indices)) { index in
-            if set.flashcards[safe: index] != nil {
+            if let flashcard = set.flashcards[safe: index],
+               flashcard != nil &&
```

Comment:
This line is responsible for an error:

```
warning: comparing non-optional value of type 'Flashcard' to 'nil' always returns true
```

It could be changed back to

```swift
set.flashcards[safe: index] != nil &&
```

I'm not sure if it can be left out. I'm unwrapping the flashcard one line above, so this check should be obsolete now, right?

Reply:
Yes, the `if let flashcard = set.flashcards[safe: index]` assigns a nonoptional value to `flashcard` so `flashcard` _cannot_ be `nil` (because of its nonoptional type) where you are checking it. So you could remove the `flashcard != nil` check.